### PR TITLE
feat: cosmos balance chart adjustments for delegation

### PR DESCRIPTION
**Pull request by elmutt**
## Description

- Include total cosmos delegation balance in charts
- Ignore chart balance changes that are a result of delegation & undelegation

## Pull Request Type

- [ ] :bug: Bug fix (Non-breaking Change: Fixes an issue)
- [ ] :hammer_and_wrench: Chore (Non-breaking Change: Doc updates, pkg upgrades, typos, etc..)
- [x] :nail_care: New Feature (Breaking/Non-breaking Change)

## Risk

Charts could look wrong if we screwed things up

## Testing

- Look at chart of an account with cosmos delegations & undelegations. Delegation balance should be reflected in the chart balance

### CODE REVIEW

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:1]:
```suggestion
```

**Comment by 0xApotheosis** [comment]
@elmutt add the PR template to the PR description please, ser.

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (gomesalexandre) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:347]: As we've discussed yesterday that accountSpecifiers will probably need to be revamped in the future to be the same everywhere (we use non-consistent terminology to define an "account specifier" and that's confusing), could we use this new `selectPubkeyishByChainId` selector for now? https://github.com/shapeshift/web/pull/1357/files#diff-bcccd934c3fc099ae907180ecdaa60f38a99865289682fc66896ddba29b6e817R14

Then l.334-348 can go in favor of selecting a la:
```
  const accountSpecifiersForChainId = useAppSelector(state =>
    selectPubkeyishByChainId(state, asset?.caip2)
  )
  const accountSpecifier = accountSpecifiersForChainId?.[0]
```

FYI @0xdef1cafe ^

File: src/state/slices/stakingDataSlice/selectors.ts
Comment (gomesalexandre) [inline src/state/slices/stakingDataSlice/selectors.ts:50]: Note: When @cjthompson asset-service metadata changes will be ready, this could be surfaced directly from the asset itself

**Review by gomesalexandre — APPROVED** [review]
LGTM, just added some comments about `accountSpecifiers` and `denom` to make it clear the direction we're heading with these, as it can be confusing for someone that doesn't have context

**Comment by elmutt** [comment]
> @elmutt add the PR template to the PR description please, ser.

done

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (elmutt) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:347]: Im down to make this change after the other PR is merged but don't want to hold up this PR waiting for it

Comment (cjthompson) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:346]: wtf is `pubkeyish`?

Comment (elmutt) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:346]: I agree "pubkeyish" needs more documentation about why its needed and how its used with the selectors.

One issue I have with the new staking selectors is they are named ending in "AccountSpecifier" but its a not the same "AccountSpecifier" we use other places. Can be very confusing and might be worth renaming

File: src/hooks/useBalanceChartData/cosmoUtils.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/cosmoUtils.ts:29]:
```suggestion
      : bnOrZero(totalCosmosStaked)
```

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:260]: do this on a single line after 256 and just return, don't make a christmas tree unnecessarily

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:355]: shouldn't have to optchain on this selector, it should do that logic and always return a string

File: src/state/slices/stakingDataSlice/selectors.ts
Comment (0xdef1cafe) [inline src/state/slices/stakingDataSlice/selectors.ts:53]:
```suggestion
    ) && '0'
```

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:346]: agree that the terminology is obtuse and i'll take the blame for it.

what we're doing with `eip155:1:0xdef1cafe` is not strictly caip10, as it doesn't work for bitcoin, as we're calling unchained with x/y/zpub

they're not strictly pubkeys either

`pubkeyish` is something @gomesalexandre and i spitballed as better than `AccountSpecifier` but that's clearly not popular either

we should probably refer to the x/y/zpub or eth/cosmos account as an `AccountSpecifier`, and the full `eip155:1:0x....` type strings as `AccountId`s

File: src/state/slices/stakingDataSlice/selectors.ts
Comment (0xdef1cafe) [reply src/state/slices/stakingDataSlice/selectors.ts:50]: @gomesalexandre what are you referring to here?

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:333]: i hate that we're doing this stuff in what is essentially the view layer.

we really need to settle on a single source of the truth for supported chains

getting it from here is *better*, but still bad, and this should come from chain adapters

https://github.com/shapeshift/web/blob/develop/src/state/slices/portfolioSlice/utils.ts#L27

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [inline src/hooks/useBalanceChartData/useBalanceChartData.ts:339]: https://github.com/shapeshift/web/blob/develop/src/context/TransactionsProvider/TransactionsProvider.tsx#L39

this logic is now duplicated in at least 3 places i know of, can we please move it to the portfolioSlice utils

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:346]: > Im down to make this change after the other PR is merged but don't want to hold up this PR waiting for it

it's like moving from one file to another?

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (elmutt) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:333]: I would argue using toCaip2 is just as clear as importing it from portfolio slice

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (elmutt) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:339]: The logic you linked in TransactionProvider looks different than what we have

File: src/state/slices/stakingDataSlice/selectors.ts
Comment (gomesalexandre) [reply src/state/slices/stakingDataSlice/selectors.ts]: We had previously discussed about asset-service potentially handling denoms as metadata. For now however, we will go with web not having knowledge of denoms at all and just using `assetId` as a discriminator, as we have currently

File: src/hooks/useBalanceChartData/cosmoUtils.ts
Comment (cjthompson) [inline src/hooks/useBalanceChartData/cosmoUtils.ts:24]: move to above line 8

**Review by theoboldfrazier — APPROVED** [review]

File: src/hooks/useBalanceChartData/cosmoUtils.ts
Comment (cjthompson) [inline src/hooks/useBalanceChartData/cosmoUtils.ts:1]: Rename this file to `cosmosUtils` instead of `cosmoUtils` for consistency

**Review by cjthompson — COMMENTED** [review]
please add tests

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (0xdef1cafe) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:338]: it's the same idea - my requested change still stands.

**Review by 0xdef1cafe — CHANGES_REQUESTED** [review]

File: src/hooks/useBalanceChartData/useBalanceChartData.ts
Comment (DaoDev44) [reply src/hooks/useBalanceChartData/useBalanceChartData.ts:340]: All the places where I find this logic, its slightly different enough to keep them separate at this time. However, this brings up a bigger problem where `accountSpecifiers` can mean a `CAIP10` or `pubKey` depending on where it's being defined. Here is an example of what I'm talking about.
https://github.com/shapeshift/web/blob/chart_cosmos_fixes2/src/state/slices/accountSpecifiersSlice/selectors.ts#L17
I feel like this is something that can (and needs) to be cleaned up, but possibly could be done in a follow up PR. Thoughts? @0xdef1cafe

**Review by 0xdef1cafe — APPROVED** [review]

**Pull request merged**
